fix(calling): failover immediately after 4 keepalive failures for cc flow

rarajes2 opened the pull request:
<!--
Hey there,\
Thank you for taking the time to improve our code! 🙂\
Please let us know why this change is necessary and what testing you have performed. \
This ensures our reviewers understand the impact of your change. \

**IMPORTANT**
FAILING TO FILL OUT THIS TEMPLATE WILL RESULT IN REJECTION OF YOUR PULL REQUEST
This is for compliance purposes with FedRAMP program.
-->

# COMPLETES #< [SPARK-569580](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-569580) >

## This pull request addresses

Failover to secondary mobius server immediately after 4 keep alive failure for contact center flow.

Here we have introduced a new variable/flag to check we need to failover immediately or not. That flag gets set only when the flow is contact centre and keepalive is failing. It gets reset back when the immediate failover logic runs.

## by making the following changes

Able to failover to backup mobius server immediately after 4 keepalive failure.

**Note**: Please check the [HAR](https://jira-eng-gpk2.cisco.com/jira/secure/attachment/1324962/failover-after-keepalive-public-repo.har) file attached on the JIRA.

### Change Type

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios were tested

Tested on the samples page by blocking the primary server domain and after 4 keepalive failure registration switched to secondary server.

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

Shreyas281299 reviewed (approved):
LGTM

sreenara commented on packages/calling/src/CallingClient/registration/register.test.ts:
```diff
@@ -732,7 +732,7 @@ describe('Registration Tests', () => {
       expect(reg.keepaliveTimer).not.toBe(timer);
 
-      webex.request.mockRejectedValueOnce(failurePayload).mockResolvedValue(successPayload);
+      webex.request.mockResolvedValue(successPayload);
```
How does this change ensure we are failing over immediately? I'm not clear about how this statement change covers the new flow.

rarajes2 replied:
The failure payload was for the primary server try but now we don't try primary server for the cc flow when it comes from the keepalive failure, here we show the success for the secondary server immediately.

sreenara reviewed (changes requested):
Please add some more details to the description on why we are using a new variable to control this immediate failover. That'll help others to return to this PR in the future and understand the change better.

sreenara reviewed (commented):
> Tested on the samples page by blocking the primary server domain and after 4 keepalive failure registration switched to secondary server.

@rarajes2 in the HAR file 'failover-after-keepalive-public-repo.har', it looks like the SDK tries to register with the primary twice before failing over to secondary. Why is that happening?

rarajes2 commented:
> > Tested on the samples page by blocking the primary server domain and after 4 keepalive failure registration switched to secondary server.
> 
> @rarajes2 in the HAR file 'failover-after-keepalive-public-repo.har', it looks like the SDK tries to register with the primary twice before failing over to secondary. Why is that happening?

@sreenara Fixed that. It was happening because we have logic to reregister (a function) after keepalive failures which first tries to attempt primary server then starts the failover. Updated the logic in both the PR and also updated JIRA with latest HAR files.
